samza-core/src/main/java/org/apache/samza/job/model/ContainerModel.java

```diff
@@ -87,6 +103,6 @@ public boolean equals(Object obj) {
   }
 
   public int compareTo(ContainerModel other) {
-    return containerId - other.getContainerId();
+    return processorId.compareTo(other.getProcessorId());
```

Comment:
Wondering if we should check for null processorId here and handle comparisons accordingly? That way, we don't crash with an NPE. The `hashCode` does seem to handle null values.